src/main/java/com/desiato/puresynth/configurations/AuthTokenFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.desiato.puresynth.configurations;
+
+import com.desiato.puresynth.services.AuthenticationService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.Collections;
+
+@Component
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private AuthenticationService authService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String token = request.getHeader("Auth-Token");
+
+        if (request.getRequestURI().equals("/api/login") || request.getRequestURI().equals("/api/user")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        if (token == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            response.getWriter().write("Unauthorized");
+            return;
+        }
+
+        try {
+            if (!authService.isUserAuthenticated(token)) {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                response.getWriter().write("Unauthorized: Invalid token");
+                return;
+            }
+
+            UserDetails userDetails = authService.loadUserByToken(token);
+            if (userDetails == null) {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                response.getWriter().write("Unauthorized: Invalid token");
+                return;
+            }
+
+            UsernamePasswordAuthenticationToken authentication =
+                    new UsernamePasswordAuthenticationToken(userDetails, null, Collections.emptyList());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+
+            filterChain.doFilter(request, response);
+        } catch (Exception e) {
```

Comment:
Why are you catching all `Exception`?

Reply:
Would you suggest to create a custom exception?

Reply:
With `Exception` you will catch everything - even Exceptions thrown by Spring or other libraries. If possible, you should only catch what you can handle

Reply:
Authentication or InvalidToken exception